src/audio_core/hle/mediandk_decoder.cpp

```diff
@@ -0,0 +1,214 @@
+        AMediaCodec_delete(mDecoder);
+        return false;
+    }
+
+    AMediaFormat_delete(format);
+    mNumChannels = num_channels;
+    mSampleRate = sample_rate;
+    return true;
+}
+
+std::optional<BinaryResponse> MediaNDKDecoder::Impl::ProcessRequest(const BinaryRequest& request) {
+    if (request.codec != DecoderCodec::AAC) {
+        LOG_ERROR(Audio_DSP, "FDK AAC Decoder cannot handle such codec: {}",
+                  static_cast<u16>(request.codec));
+        return {};
+    }
+
+    switch (request.cmd) {
+    case DecoderCommand::Init: {
+        return Initalize(request);
+    }
+    case DecoderCommand::Decode: {
+        return Decode(request);
+    }
+    case DecoderCommand::Unknown: {
+        BinaryResponse response;
+        std::memcpy(&response, &request, sizeof(response));
+        response.unknown1 = 0x0;
+        return response;
+    }
+    default:
+        LOG_ERROR(Audio_DSP, "Got unknown binary request: {}", static_cast<u16>(request.cmd));
+        return {};
+    }
+}
+
+std::optional<BinaryResponse> MediaNDKDecoder::Impl::Decode(const BinaryRequest& request) {
+    BinaryResponse response;
+    response.codec = request.codec;
+    response.cmd = request.cmd;
+    response.size = request.size;
+    response.num_channels = 2;
+    response.num_samples = 1024;
+
+    if (request.src_addr < Memory::FCRAM_PADDR ||
+        request.src_addr + request.size > Memory::FCRAM_PADDR + Memory::FCRAM_SIZE) {
+        LOG_ERROR(Audio_DSP, "Got out of bounds src_addr {:08x}", request.src_addr);
+        return response;
+    }
+    u8* data = mMemory.GetFCRAMPointer(request.src_addr - Memory::FCRAM_PADDR);
+
+    // input
+    int timeout = 160;
+    size_t buffer_size = 0;
+    uint8_t* buffer = nullptr;
+    ssize_t buffer_index = AMediaCodec_dequeueInputBuffer(mDecoder, timeout);
+    if (buffer_index < 0) {
+        LOG_ERROR(Audio_DSP, "Failed to enqueue the input samples: {}", buffer_index);
+        return response;
+    }
+    buffer = AMediaCodec_getInputBuffer(mDecoder, buffer_index, &buffer_size);
+    if (buffer_size < request.size) {
+        return response;
+    }
+    memcpy(buffer, data, request.size);
+    AMediaCodec_queueInputBuffer(mDecoder, buffer_index, 0, request.size, 0, 0);
+
+    // output
+    AMediaCodecBufferInfo info;
+    std::array<std::vector<u16>, 2> out_streams;
+    buffer_index = AMediaCodec_dequeueOutputBuffer(mDecoder, &info, timeout);
+    switch (buffer_index) {
+    case AMEDIACODEC_INFO_TRY_AGAIN_LATER:
+        LOG_ERROR(Audio_DSP, "Failed to dequeue output buffer: timeout!");
+        break;
+    case AMEDIACODEC_INFO_OUTPUT_BUFFERS_CHANGED:
+        LOG_ERROR(Audio_DSP, "Failed to dequeue output buffer: buffers changed!");
+        break;
+    case AMEDIACODEC_INFO_OUTPUT_FORMAT_CHANGED: {
+        AMediaFormat* format = AMediaCodec_getOutputFormat(mDecoder);
+        LOG_ERROR(Audio_DSP, "output format: {}", AMediaFormat_toString(format));
```

Comment:
I don't think this needs a `LOG_ERROR`, a `LOG_WARNING` is enough to fill the bill here.

```suggestion
        LOG_WARNING(Audio_DSP, "output format: {}", AMediaFormat_toString(format));
```